src/utils/url.ts

```diff
@@ -44,7 +46,10 @@ export const generateUrl: (options: {
     return fallback;
   }
 
-  const relativePath = generatePath(route, params);
+  // NOTE: ?? {} stops TS from crying and the check makes the shorter generatePath is used
+  const relativePath = params
+    ? generatePath(route, params ?? {})
```

Comment:
is there any way we can use `ALWAYS_EMPTY_OBJECT` here? I think I had some issues using that placeholder constant in such situation, but not sure

Reply:
The comment should've explained it, but isn't clear enough. If `params` is `null` or `undefined`, the other version `generatePath(route)` is used, so the one you've commented on will never go pass the `??` operator, but TS still complains about type 🤷